hugolib/site.go

```diff
@@ -1355,7 +1356,7 @@ func (s *Site) buildSiteMeta() (err error) {
 
 	s.assembleMenus()
 
-	s.Info.LastChange = s.Pages[0].Lastmod
+	s.Info.LastChange = s.Pages.ByLastmod()[ln-1].Lastmod
```

Comment:
Where's "that other place"?

I've considered adding a simple check in `Page.update()` for `p.Lastmod.After(p.Site.LastChange)`, but that wouldn't work during watch mode due to partial rebuilds.

Once we can find an efficient way to do this, I can write tests.